src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -995,7 +995,8 @@ public static void cacheEvict(Tangle tangle) throws Exception {
                 TransactionViewModel tvm = cache.get(hash);
                 if (tvm != null) {
                     if (!tvm.getIsCacheEntryFresh()) {
-                        tvm.updateDB(tangle, tvm.getTransaction(), tvm.getHash(), "");
+                        List<Pair<Indexable, Persistable>> batch = tvm.getSaveBatch();
+                        tangle.saveBatch(batch);
```

Comment:
hmmm,
I didn't say to do this thing... now I am thinking if it makes sense :thinking: 

If it is the first time we are saving the tx, then it must be this way and you fixed a serious bug...

For the updates, not so much because the data will never change

since if I am not mistaken, you just fixed a bug so let's leave it at that

Reply:
I think you got confused...
when I told you to use `saveBatch` it was in order to save a batch of transaction view models at once

Reply:
Ah. How will `saveBatch(tvmList)` be used?

Reply:
The `saveBatch` command can do a batch write of any data.
In your other PR #1713, when you evict a certain number of txs at once, you can do it via a save batch.

Then you are doing less DB queries,  less use of JNI, and possibly leveraging RocksDB optimizations